delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/error/AuthTokenError.scala

```diff
@@ -0,0 +1,48 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.error
+
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.contexts
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.ContextValue
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.JsonLdContext.keywords
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.encoder.JsonLdEncoder
+import ch.epfl.bluebrain.nexus.delta.sdk.http.HttpClientError
+import io.circe.syntax.{EncoderOps, KeyOps}
+import io.circe.{DecodingFailure, Encoder, JsonObject}
+
+sealed abstract class AuthTokenError(reason: String) extends SDKError {
+  override def getMessage: String = reason
+}
+
+object AuthTokenError {
+
+  /**
+    * Signals that an HTTP error occurred when fetching the token
+    */
+  final case class AuthTokenHttpError(cause: HttpClientError)
+      extends AuthTokenError(s"HTTP error when requesting auth token: ${cause.reason}")
+
+  /**
+    * Signals that the token was missing from the authentication response
+    */
+  final case class AuthTokenNotFoundInResponse(failure: DecodingFailure)
```

Comment:
If you use derivation, you would not need those.
Plus the token/expiry fields are in the openid spec so they will be there

Reply:
What do you mean by derivation? I realise that the fields should be there, but this is just me dealing with the errors and giving them good names